provider/scripts/spark/offline_store_spark_runner.py

```diff
@@ -111,13 +129,73 @@ def get_code_from_file(file_path, aws_region=None):
 
             f.seek(0)
             code = dill.loads(f.read())
+    elif store_type == "azure_blob_store":
+        connection_string = credentials.get("azure_connection_string")
+        container = credentials.get("azure_container_name")
+
+        if connection_string == None or container == None:
+            raise Exception("both 'azure_connection_string' and 'azure_container_name' need to be passed in as credential")
+
+        blob_service_client = BlobServiceClient.from_connection_string(connection_string)
+        container_client = blob_service_client.get_container_client(container)
+
+        transformation_path = download_blobs_to_local(container_client, file_path, "transformation.pkl") 
+        with open(transformation_path, "rb") as f:
+            code = dill.load(f)
     else:
         with open(file_path, "rb") as f:
             code = dill.load(f)
     
     return code
 
 
+def download_blobs_to_local(container_client, blob, local_filename):
+    """
+    Downloads a blob to local to be used by pandas.
+
+    Parameters:
+        client:         ContainerClient (used to interact with Azure container)
+        blob:           str (path to blob store)
+        local_filename: str (path to local file)
+    
+    Output:
+        full_path:      str (path to local file that will be used to read by pandas)
+    """
+    
+    print(f"downloading {blob} to {local_filename}")
+    if not os.path.isdir(LOCAL_DATA_PATH):
+        os.makedirs(LOCAL_DATA_PATH)
+
+    full_path = f"{LOCAL_DATA_PATH}/{local_filename}"
+    blob_client = container_client.get_blob_client(blob)
+
+    with open(full_path, "wb") as my_blob:
+        download_stream = blob_client.download_blob()
+        my_blob.write(download_stream.readall())
+
+    return full_path
+
+
+def set_spark_configs(spark, configs):
+    """
+    This method is used to set configs for Spark. It will be mostly
+    used to set access credentials for Spark to the store. 
+    """
+
+    for key, value in configs.items():
+        spark.conf.set(key, value)
+
+
+def split_key_value(values):
+    arguments = {}
+    for value in values:
+        # split it into key and value; parse the first value
+        key, value = value.split('=', 1)
```

Comment:
Using the = sign as a delimeter may be a problem, the azure key we're using ends in two ==

Reply:
so we are splitting on the first `=`. We should be good as long as the keys don't have `=` in them. The reason I did it this way is because it is the expected norm. But can modify it if we want.

Reply:
Yeah I actually meant to delete this comment as I saw you're only doing the first, so it shouldn't be an issue